api/base/serializers.py

```diff
@@ -328,6 +350,26 @@ def __init__(self, **kwargs):
         self.link_type = kwargs.pop('link_type', 'url')
         super(JSONAPIHyperlinkedGuidRelatedField, self).__init__(read_only=True, **kwargs)
 
+    def resolve(self, resource):
+        """
+        Resolves the view for target node or target comment when embedding.
+        """
+        embed_value = resource.target._id
+        if resource.target._name == 'node':
+            view_name = 'nodes:node-detail'
+            lookup_url_kwarg = 'node_id'
+        else:
```

Comment:
Yes, I like the idea of renaming to `TargetField`. Besides the fact that `JSONAPIHyperlinkedGuidRelatedField` is verbose, the field doesn't seem to be specific to GUIDs.

As far as error to raise, I think a custom error would make sense: `InvalidTargetError`, perhaps.